components/admin_console/workspace-optimization/cta_buttons.tsx

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {useHistory} from 'react-router-dom';
+
+import {trackEvent} from 'actions/telemetry_actions';
+
+import {TELEMETRY_CATEGORIES} from 'utils/constants';
+
+import './dashboard.scss';
+
+type CtaButtonsProps = {
+    learnMoreLink?: string;
+    learnMoreText?: string;
+    actionLink?: string;
+    actionText?: React.ReactNode;
+    telemetryAction?: string;
+    actionButtonCallback?: () => void;
+};
+
+const CtaButtons = ({
+    learnMoreLink,
+    learnMoreText,
+    actionLink,
+    actionText,
+    telemetryAction,
+    actionButtonCallback,
+}: CtaButtonsProps): JSX.Element => {
+    const history = useHistory();
+
+    const getClickHandler = (id: string) => () => {
+        if (typeof actionButtonCallback === 'function') {
+            actionButtonCallback();
+        } else if (actionLink?.startsWith('/')) {
+            history.push(actionLink);
+        } else if (actionLink?.startsWith('http')) {
+            window.open(actionLink, '_blank');
+        }
+
+        if (telemetryAction) {
+            trackEvent(
+                TELEMETRY_CATEGORIES.WORKSPACE_OPTIMIZATION_DASHBOARD,
+                `workspace_dashboard_${telemetryAction}_${id}`,
```

Comment:
@sri-byte this I where we compute the tracking id.

It consists of the area ("workspace_dashboard"), the specific metric id ("telemetryAction", e.g. data-privacy) and the button id ("cta" or "learn-more").

Does this work for you?

Reply:
This should work, if I interpret correctly, is the event going to be `workspace_dashboard_data-privacy_learn-more`. 

This is the type of data I receive from track events. 

![image](https://user-images.githubusercontent.com/96433562/153275975-db82207c-2415-43f2-a55f-c7ecf716e1d5.png)

Reply:
yes, that is correct ... although, I dont know the tool, is it grouped by category?
Asking since we send the category: `workspace_optimization_dashboard` (which is the value stored in `TELEMETRY_CATEGORIES.WORKSPACE_OPTIMIZATION_DASHBOARD`) and having that duplicated seems a bit redundant

Reply:
I'm not sure about that, 0/5. We could check/test that maybe in the pdat.matterlytics.com.